### PR TITLE
Added generate tag in Cli test section

### LINKED ISSUE #553: [1.8.0] Document cli testing of policies with generate rule types
### Description

kyverno/kyverno#3456 has been merged. It needs to be documented.

### Slack discussion

_No response_

### PR DESCRIPTION
**Pull request by chetak123**
## Related issue
fixes #553

## Proposed Changes

I have added the `generatedResource` tag in test cli section along an [example](https://github.com/kyverno/policies/tree/main/best-practices/add_network_policy) 

## Checklist

- [X] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [X] I have inspected the website preview for accuracy.
- [X] I have signed off my issue.

### CODE REVIEW

**Comment by chipzoller** [comment]
@chetak123 please complete the PR issue template and use closing keywords or manually link to the issue this will close.

**Review by shubham4443 — CHANGES_REQUESTED** [review]
Hi @chetak123 thank you for opening PR for my issue!
Here are some of the changes I felt that needs to be done -
1. [This line](https://github.com/kyverno/website/blob/677f54704126df6c8f48c959e28457829b78a6ea/content/en/docs/Kyverno%20CLI/_index.md?plain=1#L697) needs to be updated. It mentions only validate and mutate policy is supported which is not the case.
2. [This line](https://github.com/kyverno/website/blob/677f54704126df6c8f48c959e28457829b78a6ea/content/en/docs/Kyverno%20CLI/_index.md?plain=1#L703) - mutate and generate rule can declare a pass

File: content/en/docs/Kyverno CLI/_index.md [hunk 1]
```diff
@@ -1052,6 +1059,104 @@ skipped mutate policy add-default-resources -> resource default/Pod/nginx-demo2
 Test Summary: 2 tests passed and 0 tests failed
 ```
 
+In the following policy test, a `generate` policy rule is applied which generates a new resource from an existing resource present in resource.yaml. To test such policy the `generatedResource` tag is applied in `results[]` array, where the new generated resource is added.
```
Comment (shubham4443) [inline content/en/docs/Kyverno CLI/_index.md:1062]: This line could be refactored - To test `generate` policy, the addition of `generatedResource` field in the `results[]` array is must which is used to test against the resource generated by the policy.

**Comment by chipzoller** [comment]
@chetak123 please help close this PR out by addressing comments. It is needed in 1.8.0 docs.

File: content/en/docs/Kyverno CLI/_index.md [hunk 2]
```diff
@@ -1052,6 +1059,104 @@ skipped mutate policy add-default-resources -> resource default/Pod/nginx-demo2
...
+    generatedResource: generatedResource.yaml
+    kind: Namespace
+    result: pass
+```
+```sh
+$ kyverno test .
+Executing deny-all-traffic...
+applying 1 policy to 1 resource... 
+
+ Note : The resource field is being deprecated in 1.8.0 release. Please provide the resources under the resources parameter as an array in the results field
```
Comment (chipzoller) [inline content/en/docs/Kyverno CLI/_index.md:1149]: Please remove this deprecation notice. I have an [open issue](https://github.com/kyverno/kyverno/issues/4590) to eliminate it in the code as well.

**Review by chipzoller — CHANGES_REQUESTED** [review]

**Comment by chetak123** [comment]
Hey @chipzoller I have made the required changes.

**Comment by chipzoller** [comment]
@chetak123 would you mind please resolving conflicts here?

**Comment by chipzoller** [comment]
@chetak123 I thank you for the PR and am incorporating your content (with attribution) in my PR #643. This was due to there being enough changes that it was easier to make them all myself.

**Pull request closed without merging**